client/src/components/markdown/RenkuMarkdownWithPathTranslation.tsx

```diff
@@ -28,36 +29,73 @@ const REF_TYPES = {
   FILE_PREV: "file", // with ! in the front
 };
 
-const getFileExtension = (file_path) => {
-  if (!file_path) return null;
-
-  if (file_path.match(/\.(.*)/) === null) return null;
-  return file_path.split(".").pop().toLowerCase();
+type FileReference = {
+  data?: { content: string; file: File; file_name: string };
+  error: boolean;
+  filePreview?: boolean;
+  iBlock: number;
+  isOpened?: boolean;
+  refPath?: string;
+  type: string;
 };
 
-const fileIsImage = (file_path) =>
-  IMAGE_EXTENSIONS.indexOf(getFileExtension(file_path)) >= 0;
+type RenkuMarkdownWithPathTranslationProps = {
+  /** The branch used to retrieve the file contents */
+  branch: string;
+  /** The GitLab client */
+  client: any; // eslint-disable-line @typescript-eslint/no-explicit-any
+  /** The file path to render */
+  filePath: string;
+  /** Is the file inside a project? */
+  insideProject: boolean;
+  /** The markdown text to display */
+  markdownText: string;
+  /** Sizes at which the UI shows a preview or not */
+  previewThreshold: FilePreviewProps["previewThreshold"];
+  /** The project id */
+  projectId: string;
+  /** The project path */
+  projectPathWithNamespace: string;
+  /** If true, render the output as a single line without line breaks */
+  singleLine?: boolean;
+  /** Any styles to apply */
+  style?: object;
+};
 
-const getFilesRefs = (markdownHTML, filePathArray) => {
-  let filesRefs = [];
-  let blockCounter = 0;
+const getFileExtension = (file_path: string | undefined) => {
+  if (file_path == null) return undefined;
 
-  const previewFiles = markdownHTML.getElementsByTagName("img");
+  if (file_path.match(/\.(.*)/) === null) return undefined;
+  const components = file_path.split(".");
+  // if there is no extension, return undefined
+  if (components.length < 2) return undefined;
+  return components.pop()?.toLowerCase();
+};
 
-  for (let file of previewFiles) {
-    let src = fixRelativePath(file.getAttribute("src"), filePathArray);
+const fileIsImage = (file_path: string | undefined) => {
+  if (file_path == null) return false;
+  const ext = getFileExtension(file_path);
+  if (ext == null) return false;
+  IMAGE_EXTENSIONS.includes(ext);
+};
 
-    if (!file.src.match(patterns.urlFilesRef)) {
-      filesRefs.push({
+const getFilesRefs = (
+  markdownHTML: HTMLDivElement,
+  filePathArray: string[]
+) => {
+  const previewFiles = markdownHTML.getElementsByTagName("img");
+  const filesRefs: FileReference[] = [...previewFiles]
```

Comment:
`.filter()` and `.map()` will not mutate the original array.

```suggestion
  const filesRefs: FileReference[] = previewFiles
```

Reply:
The method [getElementsByTagName](https://developer.mozilla.org/en-US/docs/Web/API/Element/getElementsByTagName) returns an [HTMLCollection](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCollection) (see also https://dom.spec.whatwg.org/#interface-htmlcollection), not a standard JavaScript [Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array). 

Unlike Array, HTMLCollection does not contain `filter` as part of its interface, so it is necessary to convert that object into one that understands `filter`.